providers/websockets/bybit/parse.go

```diff
@@ -0,0 +1,77 @@
+package bybit
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+	"time"
+
+	"github.com/skip-mev/slinky/providers/base/websocket/handlers"
+
+	"go.uber.org/zap"
+
+	"github.com/skip-mev/slinky/pkg/math"
+	providertypes "github.com/skip-mev/slinky/providers/types"
+	oracletypes "github.com/skip-mev/slinky/x/oracle/types"
+)
+
+// parseSubscribeResponseMessage parses a subscribe response message. The format of the message
+// is defined in the messages.go file. There are two cases that are handled:
+//
+// 1. Successfully subscribed to the channel. In this case, no further action is required.
+// 2. Error message. In this case, we attempt to re-subscribe to the channel.
+func (h *WebsocketDataHandler) parseSubscriptionResponse(resp SubscriptionResponse) ([]handlers.WebsocketEncodedMessage, error) {
+	// A response with an event type of subscribe means that we have successfully subscribed to the channel.
+	if t := Operation(resp.Op); t == OperationSubscribe && resp.Success {
+		h.logger.Info("successfully subscribed to channel", zap.String("connection", resp.ConnID))
+		return nil, nil
+	}
+
+	if t := Operation(resp.Op); t == OperationSubscribe && !resp.Success {
+		h.logger.Error("received error message", zap.String("message", resp.RetMsg))
+
+		// TODO resubscribe ?
+		return nil, nil
+	}
```

Comment:
is there anyway to derive the subscription that was created? Since the subscriptions are batched into a single request, how would this be done? Seems like the sub resp doesnt include that info?

Reply:
Nope: its very annoying:

the resp is:

```
{
    "success": true,
    "ret_msg": "subscribe",
    "conn_id": "2324d924-aa4d-45b0-a858-7b8be29ab52b",
    "req_id": "10001",
    "op": "subscribe"
}
```